### ISSUE TITLE
[cssom-view] CSSOM View Module redefines some MouseEvent attributes

### ISSUE THREAD

**Issue by lukebjerring** [w3c/csswg-drafts#4084]
MouseEvent already has all of screenX, screenY, clientX, clientY (which are also `long` type).

But, they're redefined in the partial interface here:
https://drafts.csswg.org/cssom-view/#extensions-to-the-mouseevent-interface

Found while tweaking the idlharness in https://github.com/web-platform-tests/wpt/pull/12231

**Comment by jdm** [comment]
This prevents automated testing of the MouseEvent interface in idlharness.js for any test that includes cssom-view.idl because it causes validation to fail, and that's very disappointing.

**Comment by lukebjerring** [comment]
@jdm - not sure that it prevents anything, just shows up as a failure (or several).
screen-orientation, media-capabilities and cssom-view reference the cssom-view IDL.

https://wpt.fyi/results/?label=master&label=experimental&aligned&q=%28cssom-view%7Cmedia-capabilities%7Cscreen-orientation%29%20and%20idlharness

Only cssom-view has failures, for the unique members subtest, and they're:
> FAIL message: assert_true: member screenX is unique expected true got false

Or are you referring to something else?

**Comment by jdm** [comment]
I'm referring to the fact that on [the cssom-view idlharness test](https://wpt.fyi/results/css/cssom-view/idlharness.html?label=experimental&label=master&aligned&q=%28cssom-view%7Cmedia-capabilities%7Cscreen-orientation%29%20and%20idlharness) there are only three test entries that mention MouseEvent. This is because the [MouseEvent object](https://github.com/web-platform-tests/wpt/blob/8f21bdd2710f97b928cbe909f20adcfbc072b13c/css/cssom-view/idlharness.html#L37) is disabled, but uncommenting that still does not cause the MouseEvent interface to be checked. Only fixing the validation errors allows idlharness.js to continue running tests against the interface that fails validation.

**Comment by foolip** [comment]
The double definitions of these attributes is still an issue. @smfr @emilio @garykac @travisleithead as the editors of the two specs in questions, which spec do you think should define this? I started writing a patch for CSSOM View, but noticed that some other attributes are defined in terms of these, for example `mouseEvent.x` is just an alias for `mouseEvent.clientX`. That could still be maintained even if the definition of `clientX` is in UI Events, but it made me question what the best split is. Advice?

Note that by now this isn't an issue for idlharness.js tests in WPT, because those use the IDL from https://www.npmjs.com/package/@webref/idl which has patches applied to fix problems like this.

**Comment by gsnedders** [comment]
See also: https://github.com/w3c/uievents/issues/24

**Comment by mustaqahmed** [comment]
I am proposing to conflate multiple coordinate definitions as follows (based on our compat [experience](https://github.com/w3c/pointerevents/issues/100#issuecomment-818870316) that fractional coordinates work only for non-click-like Pointer Events):
- Pointer Events spec will adopt the CSSOM-View [Extensions to the MouseEvent Interface](https://drafts.csswg.org/cssom-view/#extensions-to-the-mouseevent-interface) for specific event-type for which this works.
- CSSOM-View spec will drop the extension in favor of above.
- UI Event spec will be unchanged, effectively leaving Mouse Event coordinates as integers.

**Comment by mustaqahmed** [comment]
To avoid any confusion: while Blink's IDL has `MouseEvent` coordinates as `double`, Blink still exposes only integer values for `MouseEvent`, so the following lines produce `1` and `1.23` (and my proposed spec change above implies the same):
```
(new MouseEvent('mousedown', {screenX: 1.23})).screenX;
(new PointerEvent('pointerdown', {screenX: 1.23})).screenX;
```

**Comment by emilio** [comment]
SGTM, thanks!

**Comment by garykac** [comment]
I don't think this makes sense to live in CSSOM, and as @mustaqahmed said, we're working to define the boundary between MouseEvents (in UIEvents) and PointerEvents (in the PE spec) so that PE has enough ownership/control for the spec to be written algorithmically.

**Comment by smfr** [comment]
Are the integral coordinates in MouseEvent computed by flooring or rounding?

**Comment by mustaqahmed** [comment]
Blink [floors](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/events/mouse_event.h;drc=d7873ef2a2e7af2fc4e06e8a29d7ab4a3fdb2529;l=149-153) them to match their past `int` values.
